MSAL/src/requests/MSALInteractiveRequest.m

```diff
@@ -176,77 +104,69 @@ - (void)acquireToken:(MSALCompletionBlock)completionBlock
     }];
 }
 
+
 - (void)acquireTokenImpl:(MSALCompletionBlock)completionBlock
 {
-    NSURL *authorizationUrl = [self authorizationUrl];
+    MSIDWebviewConfiguration *config = [[MSIDWebviewConfiguration alloc] initWithAuthorizationEndpoint:_authority.authorizationEndpoint
+                                                                                           redirectUri:_parameters.redirectUri
+                                                                                              clientId:_parameters.clientId resource:nil
+                                                                                                scopes:_parameters.scopes
+                                                                                         correlationId:_parameters.correlationId
+                                                                                            enablePkce:YES];
+    config.promptBehavior = MSALParameterStringForBehavior(_uiBehavior);
+    config.loginHint = _parameters.loginHint;
+    _webviewConfig = config;
     
-    MSID_LOG_INFO(_parameters, @"Launching Web UI");
-    MSID_LOG_INFO_PII(_parameters, @"Launching Web UI with URL: %@", authorizationUrl);
-    s_currentRequest = self;
+    void (^webAuthCompletion)(MSIDWebviewResponse *, NSError *) = ^void(MSIDWebviewResponse *response, NSError *error) {
+        if ([response isKindOfClass:MSIDWebOAuth2Response.class])
+        {
+            MSIDWebOAuth2Response *oauthResponse = (MSIDWebOAuth2Response *)response;
+            
+            if (oauthResponse.authorizationCode)
+            {
+                _code = oauthResponse.authorizationCode;
```

Comment:
wouldn't any other response also have code in it?

Reply:
You are right, AADAuthResponse should also have the auth code. 
For now, there are only 2 returnable types - AAD and WPJ. Will collapse to just checking for AAD Auth response,.